src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,136 @@
+     *
+     * @param input The input string to be handled.
+     * @return The task corresponding to the input string.
+     */
+    public Task handleInput(String input) {
+        String taskType = input.split(" ")[0];
+        return switch (taskType) {
+            case "T" -> new ToDo(input.substring(4), input.split(" ")[1].equals("0"));
+            case "D" -> new Deadline(input.substring(4), input.split(" ")[1].equals("0"));
+            case "E" -> new Event(input.substring(4), input.split(" ")[1].equals("0"));
+            default -> null;
+        };
+    }
+
+    /**
+     * Loads the tasks from the file and returns the list of tasks.
+     *
+     * @return The list of tasks loaded from the file.
+     * @throws IOException If an error occurs during loading of the tasks.
+     */
+    public List<Task> load() throws IOException {
+        List<Task> tasks = new ArrayList<>();
+        File file = new File(filePath);
+        if (!file.exists()) {
+            File directory = new File("./data");
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            file = new File(directory,filePath.substring(5));
+            file.createNewFile();
+            System.out.println(" File not found. Creating new file...");
+            return tasks;
+        }
+
+        if (!isFileUncorrupted(file)) {
+            throw new IOException(" File is corrupted");
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task task = this.handleInput(line);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+        }
+
+        return tasks;
+    }
+
+    /**
+     * Saves the list of tasks to the file.
+     *
+     * @param tasks The list of tasks to be saved to the file.
+     * @throws IOException If an error occurs during saving of the tasks.
+     */
+    public void save(List<Task> tasks) throws IOException {
+        try (FileWriter fw = new FileWriter(filePath)) {
+            for (Task task : tasks) {
+                fw.write(task.writeToFile());
+                fw.write("\n");
+                fw.flush();
+            }
+        }
+    }
+
+    /**
+     * Checks if the file is corrupted.
+     *
+     * @param file The file to be checked.
+     * @return True if the file is not corrupted, false otherwise.
+     */
+    public boolean isFileUncorrupted(File file) {
+        Pattern TODO_PATTERN = Pattern.compile("^T\\s+\\d\\s+\\w+.*$");
+        Pattern EVENT_PATTERN = Pattern.compile("^E\\s+\\d\\s+\\w+.*\\s+/from\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)\\s+/to\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)$");
+
+        Pattern DEADLINE_PATTERN = Pattern.compile("^D\\s+\\d\\s+\\w+.*\\s+/by\\s+\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|30)$");
```

Comment:
Great to see that constants (for this purpose) are named with all uppercase 👍

Reply:
Thank you for spending your time to review my code!